src/Components/MediaChannelComponent.ts

```diff
@@ -0,0 +1,176 @@
+        for (const c of this.channelsArray) {
+            ids.push((<TextChannel>c).id);
+        }
+        return {
+            channels: ids
+        };
+    }
+
+    async afterLoadJSON(loadedObject: MediaComponentSave | undefined): Promise<void> {
+        if (loadedObject) {
+            for (const c of loadedObject.channels) {
+                const channel = this.djmtGuild.getGuildChannel(c) as TextChannel;
+                if (!channel) {
+                    console.error(`[MediaChannelCheck]: could not load ${c}`);
+                    continue;
+                }
+                this.channelsArray.push(channel);
+            }
+        }
+    }
+
+    async onReady(): Promise<void> {
+        return Promise.resolve(undefined);
+    }
+
+    async onGuildMemberAdd(member: GuildMember): Promise<void> {
+        return Promise.resolve(undefined);
+    }
+
+    async onMessage(args: string[], message: Message): Promise<void> {
+        await this.checkMedia(message);
+    }
+
+    async onMessageReactionAdd(messageReaction: MessageReaction, user: User): Promise<void> {
+        return Promise.resolve(undefined);
+    }
+
+    async onMessageReactionRemove(messageReaction: MessageReaction, user: User): Promise<void> {
+        return Promise.resolve(undefined);
+    }
+
+    async onMessageUpdate(oldMessage: Message, newMessage: Message): Promise<void> {
+        return Promise.resolve(undefined);
+    }
+
+    async onMessageWithGuildPrefix(args: string[], message: Message): Promise<void> {
+        const command = args?.shift()?.toLowerCase() || '';
+        // Any interactive commands should be defined in ComponentCommands.ts
+        if (command === ComponentCommands.SET_MEDIA_CHANNEL) {
+            await this.setMediaChannel(args, message);
+        } else if (command === ComponentCommands.GET_MEDIA_CHANNEL) {
+            await this.getMediaChannel(message);
+        }
+        return Promise.resolve(undefined);
+    }
+
+    async onVoiceStateUpdate(oldState: VoiceState, newState: VoiceState): Promise<void> {
+        return Promise.resolve(undefined);
+    }
+
+    /**
+     * Toggles certain channels as "media channels", where any non-media messages will be deleted.
+     * @param args A list of channels to toggle.
+     * @param message The message object of the command message.
+     * @private
+     */
+    private async setMediaChannel(args: string[], message: Message): Promise<void> {
+        if (!isAdmin(message)) {
+            await message.channel.send(`This command requires administrator permissions.`);
+            return;
+        }
+        if (args.length === 0) {
+            await message.channel.send('Toggles a channel as a media channel, or multiple channels at the same time.');
+            await message.channel.send('Command syntax: \`setmc #channel1 #channel2 #channel3 ...\`');
+            await message.channel.send('You can also use \`getmc\` to see a list of set channels.');
+        }
```

Comment:
By adding a `return` here, you can get rid of the following `else` block (this also makes sure that `await this.djmtGuild.saveJSON();` call on line 133 doesn't occur when you enter no args)

try adding more returns on the other if statements inside and there will be way less { } blocks than before
```suggestion
        if (args.length === 0) {
            await message.channel.send('Toggles a channel as a media channel, or multiple channels at the same time.');
            await message.channel.send('Command syntax: \`setmc #channel1 #channel2 #channel3 ...\`');
            await message.channel.send('You can also use \`getmc\` to see a list of set channels.');
            return
        }
```